controllers/users.js

```diff
@@ -146,6 +149,40 @@ const postUserPicture = async (req, res) => {
   }
 };
 
+/**
+ * Updates the user data
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+const updateUser = async (req, res) => {
+  try {
+    const userId = req.params.userId;
+    const { id: profileDiffId, message, ...profileDiff } = req.body;
```

Comment:
Why do we need to update id here?
And could we look into more variable names instead of profileDiffId  ---> updatedUserId / profileDetails instead of profileDiff?

Reply:
The API is part of Identity Service. The variable names are taken with account of those only. 

Profile Diff are profile changes that user requested. These changes are stored in `profileDiff` model and only updated in user document when a `super_user` approves it. A user themselves can't update their data via this API.

We are passing userId as params because `req.userData` contains the data of super_user who called it.